Objects/memoryobject.c

```diff
@@ -3343,7 +3345,7 @@ memoryiter_new(PyObject *self, int reversed)
     it->it_index = reversed ? (it->it_length - 1) : 0;
     it->it_seq = _PyMemoryView_CAST(Py_NewRef(self));
     _PyObject_GC_TRACK(it);
-    return (PyObject *)it;
+    return _PyObject_CAST(it);
```

Comment:
My understanding is that `_PyObject_CAST` is reserved for macros, as it deals with properly casting on C++. But since this won't be seen by C++, it's fine to just use `(PyObject *)` (that's what we use in most C code, anyway).